### ISSUE TITLE
Error on running test.py: "Segmentation fault: 11" on Mac 12.5 (intel)

### ISSUE THREAD

**Issue by mzettersten** [yoterel/icatcher_plus#6]
Hey

Thanks for the excellent and clear documentation for this project!

In testing out the pipeline on my local machine (Macbook Pro 2018, OS 12.5) for a (single) Lookit video of my own, I get the following output and error when running test.py:

`INFO:root:using the following values for per-channel mean: [0.485, 0.456, 0.406]
INFO:root:using the following values for per-channel std: [0.229, 0.224, 0.225]
Segmentation fault: 11`

The minimal command to produce this error is (in my icatcher conda environment):
`python test.py path/to/video path/to/model/icatcher+_lookit.pth --fc_model path/to/face/model/face_classifier_lookit.pth`

Some quick googling around seems to suggest this might be a memory-related error. Any pointers/ tips/ suggestions for troubleshooting? Thanks!

**Comment by mzettersten** [comment]
Oh wow, sorry, I just realized it looks like this might be a copy of #4... Looking into the discourse on that issue now.

**Comment by mzettersten** [comment]
OK, so I don't think this is necessarily the same solution, because I am using an Intel Mac and my icatcher environment has python 3.9 installed. Here are my stats:

python version: 3.9.12
CPU: Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
MAC OS Version: 12.5

Conda info:
`# Name                    Version                   Build  Channel
absl-py                   0.15.0             pyhd3eb1b0_0    anaconda
aiohttp                   3.8.1            py39hca72f7f_1    anaconda
aiosignal                 1.2.0              pyhd3eb1b0_0    anaconda
async-timeout             4.0.1              pyhd3eb1b0_0    anaconda
attrs                     21.4.0             pyhd3eb1b0_0    anaconda
blas                      1.0                         mkl    anaconda
blinker                   1.4              py39hecd8cb5_0    anaconda
bottleneck                1.3.4            py39h67323c0_0    anaconda
brotli                    1.0.9                hb1e8313_2    anaconda
brotlipy                  0.7.0           py39h9ed2024_1003    anaconda
bzip2                     1.0.8                h1de35cc_0    anaconda
c-ares                    1.18.1               hca72f7f_0    anaconda
ca-certificates           2022.4.26            hecd8cb5_0    anaconda
cachetools                4.2.2              pyhd3eb1b0_0    anaconda
certifi                   2022.6.15        py39hecd8cb5_0    anaconda
cffi                      1.15.0           py39hc55c11b_1    anaconda
charset-normalizer        2.0.4              pyhd3eb1b0_0    anaconda
click                     8.0.4            py39hecd8cb5_0    anaconda
cryptography              3.4.8            py39h2fd3fbb_0    anaconda
cycler                    0.11.0             pyhd3eb1b0_0    anaconda
dataclasses               0.8                pyh6d0b6a4_7    anaconda
ffmpeg                    4.3                  h0a44026_0    pytorch
ffmpeg-python             0.2.0                      py_0    conda-forge
fonttools                 4.25.0             pyhd3eb1b0_0    anaconda
freetype                  2.11.0               hd8bbffd_0    anaconda
frozenlist                1.2.0            py39hca72f7f_0    anaconda
future                    0.18.2           py39hecd8cb5_1    anaconda
gettext                   0.21.0               h7535e17_0    anaconda
giflib                    5.2.1                haf1e3a3_0    anaconda
gmp                       6.2.1                he9d5cce_3    anaconda
gnutls                    3.6.15               hed9c0bf_0    anaconda
google-auth               2.6.0              pyhd3eb1b0_0    anaconda
google-auth-oauthlib      0.4.1                      py_2    anaconda
grpcio                    1.42.0           py39ha29bfda_0    anaconda
icu                       58.2                 h0a44026_3    anaconda
idna                      3.3                pyhd3eb1b0_0    anaconda
importlib-metadata        4.11.3           py39hecd8cb5_0    anaconda
iniconfig                 1.1.1              pyhd3eb1b0_0    anaconda
intel-openmp              2021.4.0          hecd8cb5_3538    anaconda
joblib                    1.1.0              pyhd3eb1b0_0    anaconda
jpeg                      9e                   hca72f7f_0    anaconda
kiwisolver                1.4.2            py39he9d5cce_0    anaconda
lame                      3.100                h1de35cc_0    anaconda
lcms2                     2.12                 hf1fd2bf_0    anaconda
libcxx                    12.0.0               h2f01273_0    anaconda
libffi                    3.3                  hb1e8313_2    anaconda
libgfortran               3.0.1                h93005f0_2    anaconda
libiconv                  1.16                 hca72f7f_2    anaconda
libidn2                   2.3.2                h9ed2024_0    anaconda
libpng                    1.6.37               ha441bb4_0    anaconda
libprotobuf               3.20.1               h8346a28_0    anaconda
libtasn1                  4.16.0               h9ed2024_0    anaconda
libtiff                   4.2.0                hdb42f99_1    anaconda
libunistring              0.9.10               h9ed2024_0    anaconda
libwebp                   1.2.2                h56c3ce4_0    anaconda
libwebp-base              1.2.2                hca72f7f_0    anaconda
libxml2                   2.9.14               hbf8cd5e_0    anaconda
littleutils               0.2.2                      py_0    conda-forge
llvm-openmp               12.0.0               h0dcd299_1    anaconda
lz4-c                     1.9.3                h23ab428_1    anaconda
markdown                  3.3.4            py39hecd8cb5_0    anaconda
matplotlib                3.5.1            py39hecd8cb5_1    anaconda
matplotlib-base           3.5.1            py39hfb0c5b7_1    anaconda
mkl                       2021.4.0           hecd8cb5_637    anaconda
mkl-service               2.4.0            py39h9ed2024_0    anaconda
mkl_fft                   1.3.1            py39h4ab4a9b_0    anaconda
mkl_random                1.2.2            py39hb2f4e1b_0    anaconda
mpmath                    1.2.1            py39hecd8cb5_0    anaconda
multidict                 5.2.0            py39hca72f7f_2    anaconda
munkres                   1.1.4                      py_0    anaconda
ncurses                   6.3                  hca72f7f_2    anaconda
nettle                    3.7.3                h230ac6f_1    anaconda
numexpr                   2.8.1            py39h2e5f0a9_2    anaconda
numpy                     1.22.3           py39h2e5f0a9_0    anaconda
numpy-base                1.22.3           py39h3b1a694_0    anaconda
oauthlib                  3.2.0              pyhd3eb1b0_0    anaconda
opencv-python             4.6.0.66                 pypi_0    pypi
openh264                  2.1.1                h8346a28_0    anaconda
openssl                   1.1.1o               hca72f7f_0    anaconda
outdated                  0.2.1              pyhd8ed1ab_0    conda-forge
packaging                 21.3               pyhd3eb1b0_0    anaconda
pandas                    1.4.2            py39he9d5cce_0    anaconda
pandas-flavor             0.2.0                      py_0    conda-forge
patsy                     0.5.2            py39hecd8cb5_1    anaconda
pillow                    9.0.1            py39hde71d04_0    anaconda
pingouin                  0.5.2              pyhd8ed1ab_0    conda-forge
pip                       21.2.4           py39hecd8cb5_0    anaconda
pluggy                    1.0.0            py39hecd8cb5_1    anaconda
protobuf                  3.20.1           py39he9d5cce_0    anaconda
py                        1.11.0             pyhd3eb1b0_0    anaconda
pyasn1                    0.4.8              pyhd3eb1b0_0    anaconda
pyasn1-modules            0.2.8                      py_0    anaconda
pycparser                 2.21               pyhd3eb1b0_0    anaconda
pyjwt                     2.1.0            py39hecd8cb5_0    anaconda
pyopenssl                 21.0.0             pyhd3eb1b0_1    anaconda
pyparsing                 3.0.4              pyhd3eb1b0_0    anaconda
pysocks                   1.7.1            py39hecd8cb5_0    anaconda
pytest                    7.1.2            py39hecd8cb5_0    anaconda
python                    3.9.12               hdfd78df_1    anaconda
python-dateutil           2.8.2              pyhd3eb1b0_0    anaconda
pytorch                   1.12.1                  py3.9_0    pytorch
pytz                      2022.1           py39hecd8cb5_0    anaconda
readline                  8.1.2                hca72f7f_1    anaconda
requests                  2.27.1             pyhd3eb1b0_0    anaconda
requests-oauthlib         1.3.0                      py_0    anaconda
rsa                       4.7.2              pyhd3eb1b0_1    anaconda
scikit-learn              1.0.2            py39hae1ba45_1    anaconda
scipy                     1.7.3            py39h8c7af03_0    anaconda
seaborn                   0.11.2             pyhd3eb1b0_0    anaconda
setuptools                61.2.0           py39hecd8cb5_0    anaconda
six                       1.16.0             pyhd3eb1b0_1    anaconda
sqlite                    3.38.5               h707629a_0    anaconda
statsmodels               0.13.2           py39hca72f7f_0    anaconda
tabulate                  0.8.9            py39hecd8cb5_0    anaconda
tensorboard               2.6.0                      py_1    anaconda
tensorboard-data-server   0.6.0            py39h5896577_0    anaconda
tensorboard-plugin-wit    1.6.0                      py_0    anaconda
threadpoolctl             2.2.0              pyh0d69192_0    anaconda
tk                        8.6.12               h5d9f67b_0    anaconda
tomli                     1.2.2              pyhd3eb1b0_0    anaconda
torchvision               0.13.1                 py39_cpu    pytorch
tornado                   6.1              py39h9ed2024_0    anaconda
tqdm                      4.64.0           py39hecd8cb5_0    anaconda
typing-extensions         4.1.1                hd3eb1b0_0    anaconda
typing_extensions         4.1.1              pyh06a4308_0    anaconda
tzdata                    2022a                hda174b7_0    anaconda
urllib3                   1.26.9           py39hecd8cb5_0    anaconda
werkzeug                  2.0.3              pyhd3eb1b0_0    anaconda
wheel                     0.37.1             pyhd3eb1b0_0    anaconda
xarray                    0.20.1             pyhd3eb1b0_1    anaconda
xz                        5.2.5                hca72f7f_1    anaconda
yarl                      1.6.3            py39h9ed2024_0    anaconda
zipp                      3.8.0            py39hecd8cb5_0    anaconda
zlib                      1.2.12               h4dc903c_2    anaconda
zstd                      1.5.2                hcb37349_0    anaconda`

**Comment by yoterel** [comment]
Thanks for sharing this information !
The problem is I can't reproduce this because I am not in possession of a similar machine. I was hoping upgrading to python 3.9 would help. 

Is there any chance you can run a python debugger and tell me exactly where is the segmentation fault thrown from in the code? this would be tremendously helpful.

here is a quick and dirty way (in case you don't have an IDE like visual studio code etc):

`python -m pdb -c continue test.py`

and post here what you get when it dies.

This doesn't reproduce on Windows 10 or Linux, so please consider using a machine with these OS.

**Comment by GalRaz** [comment]
@yoterel I've been able to reproduce this issue on the same OS as @mzettersten (segmentation error) when trying to run icatcher+ locally. But I am not successful at finding the line at which it happens. `python -m pdb -c continue test.py
` doesn't give any extra info, and when we pass params through VS code instead of command line, the segmentation error disappears but there is no output. Happy to jump on a call to look at it together if you have time @yoterel

**Comment by ebbethke** [comment]
I think I got the script working by rolling back PyTorch to v1.11.0.  Testing with clean install now, I'll confirm as soon as I can.

**Comment by ebbethke** [comment]
Confirmed: specifying ```pytorch=1.11.0``` and ```torchvision=0.12.0``` in the environment.yml file with a clean install solved the segfault on MacOSX 12.5 and Windows 10 build 19042.1889.  @yoterel how would you prefer to test/close this?  Happy to help if I can.

**Comment by annikamh406** [comment]
This solved the issue for me as well on MacOS 12.5.1 with an M1 chip!

**Comment by yoterel** [comment]
Closing issue as the "bug" is not in our code base but rather in torchvisions support for this architecture.
Thank you for finding a solution !

**Comment by mzettersten** [comment]
just confirming that this solved the issue for me as well. Thanks so much for tracking this down @ebbethke & co.!!
